app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt

```diff
@@ -0,0 +1,164 @@
+
+  @Inject
+  lateinit var questionnaireResponseSyncApi: QuestionnaireResponseSyncApi
+
+  @Inject
+  lateinit var userSession: UserSession
+
+  @Inject
+  lateinit var syncInterval: SyncInterval
+
+  @get:Rule
+  val ruleChain: RuleChain = Rules
+      .global()
+      .around(ServerAuthenticationRule())
+      .around(SaveDatabaseRule())
+
+  private lateinit var questionnaireSync: QuestionnaireSync
+
+  private lateinit var questionnaireResponseSync: QuestionnaireResponseSync
+
+  private val batchSize = 3
+  private lateinit var config: SyncConfig
+
+  private val currentFacilityUuid: UUID by unsafeLazy { userSession.loggedInUserImmediate()!!.currentFacilityUuid }
+
+  @Before
+  fun setUp() {
+    TestClinicApp.appComponent().inject(this)
+
+    resetLocalData()
+
+    config = SyncConfig(
+        syncInterval = syncInterval,
+        pullBatchSize = batchSize,
+        pushBatchSize = batchSize,
+        name = ""
+    )
+
+    questionnaireSync = QuestionnaireSync(
+        syncCoordinator = SyncCoordinator(),
+        api = questionnaireSyncApi,
+        repository = questionnaireRepository,
+        lastPullToken = questionnaireLastPullToken,
+        config = config
+    )
+
+    questionnaireResponseSync = QuestionnaireResponseSync(
+        syncCoordinator = SyncCoordinator(),
+        api = questionnaireResponseSyncApi,
+        repository = questionnaireResponseRepository,
+        lastPullToken = lastPullToken,
+        config = config
+    )
+  }
+
+  private fun resetLocalData() {
+    clearQuestionnaireData()
+    clearQuestionnaireResponseData()
+    lastPullToken.delete()
+    questionnaireLastPullToken.delete()
+  }
+
+  private fun clearQuestionnaireData() {
+    appDatabase.questionnaireDao().clearData()
+  }
+
+  private fun clearQuestionnaireResponseData() {
+    appDatabase.questionnaireResponseDao().clear()
+  }
+
+  @Test
+  fun syncing_records_should_work_as_expected() {
+    // given
+    val totalNumberOfRecords = batchSize * 2 + 1
+
+    questionnaireSync.pull()
+    val monthlyScreeningReportQuestionnaire = questionnaireRepository.questionnairesByType(MonthlyScreeningReports)
+
+    Truth.assertThat(monthlyScreeningReportQuestionnaire).isNotNull()
+
+    val records = (1..totalNumberOfRecords).map {
+      val uuid = UUID.randomUUID()
+      TestData.questionnaireResponse(
+          uuid = uuid,
+          questionnaireId = monthlyScreeningReportQuestionnaire.uuid,
+          questionnaireType = monthlyScreeningReportQuestionnaire.questionnaire_type,
+          facilityId = currentFacilityUuid,
+          syncStatus = SyncStatus.PENDING
+      )
+    }
+    Truth.assertThat(records).containsNoDuplicates()
+
+    questionnaireResponseRepository.save(records)
+    Truth.assertThat(questionnaireResponseRepository.pendingSyncRecordCount().blockingFirst()).isEqualTo(totalNumberOfRecords)
+
+    // when
+    questionnaireResponseSync.push()
+    clearQuestionnaireResponseData()
+    questionnaireResponseSync.pull()
```

Comment:
Does pull change the SyncStatus of a record, or on successful push, the status gets marked as `DONE`?

Reply:
no, pull doesn't change the sync status. It's only updated on a push.